File: POO/3/3.java
public class Producto {
    private String nombre;
    private double precio;
    private int stock;
    public Producto(String nombre, double precio, int stock){
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }
    public boolean vender(int cantidad){
        if(cantidad <= 0) return false;
        if(cantidad > stock) return false;
        stock -= cantidad;
        return true;
    }
    public boolean reabastecer(int cantidad){
        if(cantidad > 0){
            stock += cantidad;
            return true;
        }
        return false;
    }
    public int getStock(){ return stock; }
    public static void main(String[] args){
        Producto p = new Producto("Jugo",5.0,10);
        System.out.println(p.vender(3) + " " + p.getStock());
        System.out.println(p.reabastecer(5) + " " + p.getStock());
        System.out.println(p.vender(20) + " " + p.getStock());
    }
}

File: POO/3/3.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock
    def vender(self, cantidad):
        if cantidad <= 0:
            return False
        if cantidad > self.stock:
            return False
        self.stock -= cantidad
        return True
    def reabastecer(self, cantidad):
        if cantidad > 0:
            self.stock += cantidad
            return True
        return False

if __name__ == "__main__":
    p = Producto("Jugo", 5.0, 10)
    ok1 = p.vender(3)
    print(ok1, p.stock)
    ok2 = p.reabastecer(5)
    print(ok2, p.stock)
    ok3 = p.vender(20)
    print(ok3, p.stock)
